llarp/link/i_link_manager.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef LLARP_I_LINK_MANAGER_HPP
```

Comment:
fwiw the convention I've tried to stick with is `AbstractX` (e.g. `AbstractRouter`)

Reply:
also introduce a `.cpp` to go with this, which just `#includes` this file. this just verifies this header file compiles in a standalone fashion.

Reply:
@majestrate has used `IClassName` and I guess he prefers that (as do I) so I didn't change that.  Not saying I won't, just haven't yet pending further discussion/convincing.

empty includes done.